code/processes/wdb.q

```diff
@@ -372,13 +377,28 @@ endofdaysort:{[dir;pt;tablist;writedownmode;mergelimits;hdbsettings]
 
 /-function to send reload message to rdbs/hdbs
 reloadproc:{[h;d;ptype]
-	.wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b]each reloadorder];
-	$[eodwaittime>0;
-		{[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]}[({@[`. `reload;x;()]; (neg .z.w)(`.wdb.handler;1b); (neg .z.w)[]};d);h;ptype];
-		@[h;(`reload;d);{[ptype;e] .lg.e[`reloadproc;"failed to reload the ",string[ptype],".  The error was : ",e]}[ptype]]
-	];
-	.lg.o[`reload;"the ",string[ptype]," has been successfully reloaded"];
-	}
+        /-default message to be inserted into reload summary table (.wdb.reloadsummary) will be changed if reload fails
+        .wdb.reloadstatus:(ptype,1b,`$"successfully reloaded");
+        /-count of processes to be reloaded 
+        .wdb.countreload:count[raze .servers.getservers[`proctype;;()!();1b;0b]each reloadorder];
+        /-defining lambdas to be executed in conditional statement 
+        /-async call back function executed when eodwaittime>0
+        sendfunc:{[x;y;ptype].[{neg[y]@x};(x;y);{[ptype;x].lg.e[`reloadproc;"failed to reload the ",string[ptype]];'x}[ptype]]};
+        /-reload function sent to processes by sendfunc in order to call process to reload. Calls process to reload, if process fails to reload
+        /-execute lambda to set .wdb.reloadstatus to an error message. Then message wdb to execute wdb.handler function.
+        reloadfunc:{[d;ptype]@[`. `reload;d;
+        /-lambda to execute if reload fails, sets .wdb.reloadstatus to error message and logs error in process error log
+                         {[ptype;e](neg .z.w)(set;`.wdb.reloadstatus;(ptype;0b;`$"reload failed with error: ",e));
+                         .lg.e[`reloadproc;"failed to reload ",string[ptype]," from .wdb.reloadproc call. The error was : ",e]}[ptype]];
+        /-send message to wdb to execute .wdb.handler function
+                         (neg .z.w)(`.wdb.handler;`); (neg .z.w)[]};
+        $[eodwaittime>0;
+                 sendfunc[(reloadfunc;d;ptype);h;ptype];       
+		 @[h;(`reload;d);{[ptype;e] .lg.e[`reloadproc;"failed to reload the ",string[ptype],".  The error was : ",e]}[ptype]]
+        ];
+        .lg.o[`reload;string[ptype]," reload has finished"];
```

Comment:
I'm not sure this log message is correct (I know it's the same as it was previously). 

I think that if eodwaittime is > 0 then it should print that the reload has been sent
And if the reload is synchronous, it should print something depending on whether the reload was successful

Reply:
Will look into this now